patches/server/1056-Implement-CartographyItemEvent.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Janet Blackquill <uhhadd@gmail.com>
+Date: Sun, 7 Apr 2024 16:52:42 -0400
+Subject: [PATCH] Implement CartographyItemEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index 860d33ac56680babd87358b1d7d2216dac3f7865..5eb8bcb43c019384bfdf21677446deb770ee87db 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -3101,6 +3101,19 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
+                             }
+                         }
+ 
++                        // Paper start - cartography item event
++                        if (packet.getSlotNum() == 2 && top instanceof org.bukkit.inventory.CartographyInventory) {
```

Comment:
Same for the `2` here. You can also use the Java 14 instanceof (e.g. `top instanceof CartographyInventory cartInv`)